src/test/object.test.ts

```diff
@@ -270,6 +271,54 @@ describe('testing POST object via multipart upload', () => {
       })
     )
   })
+
+  test('should not add row to database if upload fails', async () => {
```

Comment:
Added a test by mocking the upload object function and always causing it to throw.

The test depends on the `list` endpoint to ensure that the row in `storage.objects` is cleaned up. The alternative would be to manually check the PostgREST server, but I thought that might be too much work to include here.

Reply:
We have postgrest running anyway right, can you use postgrest-js to check the database so that we aren't relying on list endpoint's correctness?

Reply:
Updated to use `postgrest-js` to ensure that the row does not exist.